pkg/kube/kube_handler.go

```diff
@@ -963,13 +944,16 @@ func (a *userHandler) ValidateAppRepository(appRepoBody io.ReadCloser, requestNa
 	// Split body parsing to a different function for ease testing
 	appRepo, cli, err := a.getValidationCli(appRepoBody, requestNamespace, a.kubeappsNamespace)
 	if err != nil {
-		return nil, err
+		return &ValidationResponse{
+			Code:    400,
+			Message: err.Error(),
+		}, nil
```

Comment:
This is the last change that the tests brought up: if the request is for a global app repo but has a docker secret, an error is returned that is most definitely a validation error (ie. should be returned as the validation response). As it was the only error returned by `getValidationCli` I've just updated to return the validation response here on error.